src/transformers/modeling_utils.py

```diff
@@ -2995,6 +3054,15 @@ def from_pretrained(
                 kwargs["skip_keys"] = model._skip_keys_device_placement
             dispatch_model(model, **kwargs)
 
+        if quantization_method_from_args == QuantizationMethod.GPTQ:
+            if quantization_config.tokenizer is None:
+                quantization_config.tokenizer = pretrained_model_name_or_path
+            if cls.main_input_name != "input_ids":
+                raise RuntimeError("We can only quantize pure text model.")
+            quantizer.quantize_model(model, quantization_config.tokenizer)
+            model.config.quantization_config = GPTQConfig.from_dict(quantizer.to_dict())
+            model.is_gptq_quantized = True
```

Comment:
Can we have a unique `is_quantized` attribute (common to all methods of quantization) and then a a `model.quantization_method` or something similar that tells us the method used?

Reply:
Yes, we should do that. I did the modification you suggested. There should not be issues with backward compatibility. All the bnb tests passed on peft and accelerate.